src/gtl/services/dataTableService.ts

```diff
@@ -0,0 +1,102 @@
+import {IDataTableService, IRowsColsResponse} from '../interfaces/dataTable';
+
+/**
+ * @memberof miqStaticAssets.gtl
+ * @ngdoc service
+ * @name DataTableService
+ * @param $http {ng.IHttpService} http service for fetching rows and columns of data table.
+ * @param MiQEndpointsService {Object} service which holds endpoints to each data store.
+ */
+export default class DataTableService implements IDataTableService {
+  private columns: any[];
+  private rows: any[];
+  private settings: any;
+  /*@ngInject*/
+  constructor(private $http: ng.IHttpService, private MiQEndpointsService: any) {}
+
+  /**
+   * Public method for fetching data from url combined from `MiQEndpointsService.rootPoint` and
+   * `MiQEndpointsService.endpoints.listDataTable`. Result will be promise with type `IRowsColsResponse`. Which is
+   * ```javascript
+   * interface IRowsColsResponse {
+   *  rows: any[];
+   *  cols: any[];
+   *  settings: ITableSettings;
+   * }
+   * ```
+   * @methodOf miqStaticAssets.gtl
+   * @memberof DataTableService
+   * @function retrieveRowsAndColumnsFromUrl
+   * @returns {ng.IPromise<IRowsColsResponse>} promise with type `IRowsColsResponse`.
+   */
+  public retrieveRowsAndColumnsFromUrl(modelName?: string,
+                                       activeTree?: string,
+                                       currId?: string): ng.IPromise<IRowsColsResponse> {
+    return this.fetchData(DataTableService.generateConfig(modelName, activeTree, currId))
+      .then(responseData => {
+        this.columns = responseData.data.data.head;
+        this.rows = responseData.data.data.rows;
+        this.settings = responseData.data.settings;
+        return {
+          cols: this.columns,
+          rows: this.rows,
+          settings: responseData.data.settings
+        };
+      }
+    );
+  }
+
+  /**
+   * Method which will do actual http get request using $http service.
+   * @param config which contains config params.
+   * @returns {IHttpPromise<any>} promise for later data filtering.
+   */
+  private fetchData(config: any): ng.IPromise<any> {
+    return this.$http.get(
+      this.MiQEndpointsService.rootPoint + this.MiQEndpointsService.endpoints.listDataTable,
+      config
+    );
+  }
+
+  /**
+   * Static function which will generate http get config from given variables.
+   * @param modelName string with name of model.
+   * @param activeTree string with active tree.
+   * @param currId ID of current item.
+   * @returns {{params: {}}} config object with params set.
+   */
+  private static generateConfig(modelName?: string, activeTree?: string, currId?: string) {
+    let config = {params: {}};
+    _.assign(config.params, DataTableService.generateModelConfig(modelName));
```

Comment:
We probably need to figure out the priority of libraries to use, when there is overlapping functionality. This statement is totally correct but should one use Object.assign(), _.assign(), [or  even angular.merge, angular.extend].

Anyway, we probably need a standard approach to this so that each of of doesn't do it differently.
@himdel @martinpovolny wdyt?